### PR TITLE
Feat: Review Spelling of Listing Code

### PR DESCRIPTION
**Pull request by danOIntellitect**
## Summary

Added a spell check editor config along with a exlusion.dic file that contains the strings that we determine are spelled correctly. Then I went through the source code and corrected the misspelled words.

Fixes 1618 from the other repo : https://github.com/IntelliTect-dev/EssentialCSharp.Tooling/issues/1618

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
```
Comment (Keboo) [inline .editorconfig]: nit, I would start with the default editorconfig. You can generate one by running `dotnet new editorconfig` from the root of the repo.

File: exclusion.dic [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+csproj
+Usings
+warnaserror
+lpfl
+cpuid
+retq
+vmax
+decryptor
+Abidal
+Autoproperty
```
Comment (Keboo) [inline exclusion.dic]: Can we drop this and fix the usage to be "AutoProperty"?

File: exclusion.dic [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+Dever
+Stokesbary
+Heavey
+enumtest
+Equatable
+Fireswamp
+spokane
+urls
+iassist
+worldvision
```
Comment (Keboo) [inline exclusion.dic]: Is it possible to drop this and fix usage to be `worldVision`?

File: exclusion.dic [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+Deconstructors
+Hartfelt
+Lindherst
+Parksdale
+Monteray
+inigo
+Machava
+Uncompress
+firstname
+lastname
```
Comment (Keboo) [inline exclusion.dic]: Can we drop these in favor of change usage to have a capital N for name?

File: src/Chapter01/Listing01.22.SampleCILOutput.cs [hunk 5]
```diff
@@ -58,7 +58,7 @@ .maxstack 8
     } // End of method System.Void HelloWorld.Program::Main(System.String[])
 
 
-  .method public hidebysig specialname rtspecialname instance void .ctor() cil managed
+  .method public hidebysig specialName rtspecialname instance void .ctor() cil managed
```
Comment (Keboo) [inline src/Chapter01/Listing01.22.SampleCILOutput.cs]: Sorry I was wrong when I told you to make this change, I did not realize it was an CIL identifier. This should be all lowercase.

**Review by Keboo — APPROVED** [review]

File: exclusion.dic (hunk 3 above)
Comment (danOIntellitect) [reply exclusion.dic]: image
Here is the usage, I would argue to keep it worldvision since it is a link. But capital V still works for the link

Comment (danOIntellitect) [reply exclusion.dic]: I made the change in the most recent commit, but did it separately if we want to roll it back

**Review by Keboo — APPROVED** [review]

**Pull request merged**